### PR TITLE
Adds derivation for RawPointerConverter trait and removes generic impl

### PR DESCRIPTION
**Pull request by anthonyreinettesonos**

### CODE REVIEW

File: ffi-convert/src/conversions.rs [hunk 1]
```diff
@@ -107,22 +123,32 @@ pub struct UnexpectedNullPointerError;
...
 /// `into_raw_pointer` method (and is thus unsafe as we don't have any way to get insurance of that
 /// from the compiler).
 ///
 /// The `from_raw_pointer` effectively takes back ownership of the pointer. If you didn't create the
 /// pointer yourself, please use the `as_ref` method on the raw pointer to borrow it
-///
-/// A generic implementation of this trait exist for every struct, it will use a `Box` to create the
-/// pointer. There is also a special implementation available in order to create a
-/// `*const libc::c_char` from a CString.
 pub trait RawPointerConverter<T>: Sized {
```
Comment (fredszaq) [inline ffi-convert/src/conversions.rs:140]: we probably wanna add mut variants here as well

**Review by fredszaq — CHANGES_REQUESTED** [review]

File: ffi-convert/src/conversions.rs [hunk 2]
```diff
@@ -107,22 +131,48 @@ pub struct UnexpectedNullPointerError;
...
-        T::from_raw_pointer(input).map(|_| ())
+        Self::from_raw_pointer(input).map(|_| ())
+    }
+
+    unsafe fn drop_raw_pointer_mut(input: *mut T) -> Result<(), UnexpectedNullPointerError> {
+        Self::from_raw_pointer_mut(input).map(|_| ())
+    }
+}
+
+pub fn convert_into_raw_pointer<T>(pointee: T) -> *const T {
```
Comment (fredszaq) [inline ffi-convert/src/conversions.rs:156]: as this is public we should document it (or put a #[doc(hidden)] on it)

File: ffi-convert/src/conversions.rs [hunk 3]
```diff
@@ -107,22 +131,48 @@ pub struct UnexpectedNullPointerError;
...
+    Box::into_raw(Box::new(pointee)) as _
+}
+
+pub fn convert_into_raw_pointer_mut<T>(pointee: T) -> *mut T {
```
Comment (fredszaq) [inline ffi-convert/src/conversions.rs:161]: as this is public we should document it (or put a #[doc(hidden)] on it)

File: ffi-convert/src/conversions.rs [hunk 4]
```diff
@@ -107,22 +131,48 @@ pub struct UnexpectedNullPointerError;
...
+    Box::into_raw(Box::new(pointee))
+}
+
+pub unsafe fn take_back_from_raw_pointer<T>(
```
Comment (fredszaq) [inline ffi-convert/src/conversions.rs:166]: as this is public we should document it (or put a #[doc(hidden)] on it)

File: ffi-convert/src/conversions.rs [hunk 5]
```diff
@@ -107,22 +131,48 @@ pub struct UnexpectedNullPointerError;
...
+    input: *const T,
+) -> Result<T, UnexpectedNullPointerError> {
+    take_back_from_raw_pointer_mut(input as _)
+}
+
+pub unsafe fn take_back_from_raw_pointer_mut<T>(
```
Comment (fredszaq) [inline ffi-convert/src/conversions.rs:173]: as this is public we should document it (or put a #[doc(hidden)] on it)

**Review by fredszaq — CHANGES_REQUESTED** [review]

File: ffi-convert/src/conversions.rs (hunk 2 above)
Comment (anthonyreinettesonos) [reply ffi-convert/src/conversions.rs:156]: Correct ! I will hide it from the documentation though.

**Comment by fredszaq** [comment]
Also regarding the documentation, we may want to add the derive to the examples doc at the root of the crate

File: ffi-convert/src/lib.rs [hunk 6]
```diff
@@ -86,10 +86,15 @@
 //! }
 //! ```
 //!
-//! You may have noticed that you have to derive the CDrop trait.
+//! You may have noticed that you have to derive two traits : the CDrop trait and the RawPointerConverter trait.
```
Comment (fredszaq) [inline ffi-convert/src/lib.rs:90]: in the visible code in the example RawPointerConverter is not derived

**Review by fredszaq — CHANGES_REQUESTED** [review]

**Pull request merged**